vendors/swedbank-pay-core-library/src/SwedbankPay/Core/Library/OrderAction.php

```diff
@@ -0,0 +1,573 @@
+<?php
+
+namespace SwedbankPay\Core\Library;
+
+use SwedbankPay\Core\Api\Response;
+use SwedbankPay\Core\Api\Transaction;
+use SwedbankPay\Core\Api\TransactionInterface;
+use SwedbankPay\Core\Exception;
+use SwedbankPay\Core\Order;
+use SwedbankPay\Core\OrderInterface;
+
+trait OrderAction
+{
+    /**
+     * Can Capture.
+     *
+     * @param mixed $orderId
+     * @param float|int|null $amount
+     *
+     * @return bool
+     */
+    public function canCapture($orderId, $amount = null)
+    {
+        // Check the payment state online
+        $order = $this->getOrder($orderId);
+
+        // Fetch payment info
+        try {
+            $result = $this->fetchPaymentInfo($order->getPaymentId());
+        } catch (\Exception $e) {
+            // Request failed
+            return false;
+        }
+
+        return isset($result['payment']['remainingCaptureAmount']) && (float)$result['payment']['remainingCaptureAmount'] > 0.1;
+    }
+
+    /**
+     * Can Cancel.
+     *
+     * @param mixed $orderId
+     * @param float|int|null $amount
+     *
+     * @return bool
+     */
+    public function canCancel($orderId, $amount = null)
+    {
+        // Check the payment state online
+        $order = $this->getOrder($orderId);
+
+        // Fetch payment info
+        try {
+            $result = $this->fetchPaymentInfo($order->getPaymentId());
+        } catch (\Exception $e) {
+            // Request failed
+            return false;
+        }
+
+        return isset($result['payment']['remainingCancellationAmount']) && (float)$result['payment']['remainingCancellationAmount'] > 0.1;
+    }
+
+    /**
+     * Can Refund.
+     *
+     * @param mixed $orderId
+     * @param float|int|null $amount
+     *
+     * @return bool
+     */
+    public function canRefund($orderId, $amount = null)
+    {
+        $order = $this->getOrder($orderId);
+
+        if (!$amount) {
+            $amount = $order->getAmount();
+        }
+
+        // Should has payment id
+        $paymentId = $order->getPaymentId();
+        if (!$paymentId) {
+            return false;
+        }
+
+        // Should be captured
+        // @todo Check payment state
+
+        // Check refund amount
+        $result = $this->fetchTransactionsList($order->getPaymentId());
+
+        $refunded = 0;
+        foreach ($result['transactions']['transactionList'] as $key => $transaction) {
+            if ($transaction['type'] === 'Reversal') {
+                $refunded += ($transaction['amount'] / 100);
+            }
+        }
+
+        $possibleToRefund = $order->getAmount() - $refunded;
+        if ($amount > $possibleToRefund) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Capture.
+     *
+     * @param mixed $orderId
+     * @param mixed $amount
+     * @param mixed $vatAmount
+     *
+     * @return Response
+     * @throws Exception
+     */
+    public function capture($orderId, $amount = null, $vatAmount = 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method capture() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)